sensing/pointcloud_preprocessor/src/blockage_diag/blockage_diag_nodelet.cpp

```diff
@@ -132,114 +184,184 @@ void BlockageDiagComponent::filter(
+            full_size_depth_map.at<uint16_t>(p.ring, horizontal_bin) =
+              UINT16_MAX - distance_coefficient * p.distance;
+          } else if (lidar_model_ == "PandarQT") {
+            full_size_depth_map.at<uint16_t>(vertical_bins - p.ring - 1, horizontal_bin) =
+              UINT16_MAX - distance_coefficient * p.distance;
+          }
         }
       }
     }
-    lidar_depth_map.convertTo(lidar_depth_map, CV_8UC1, 1.0 / 100.0);
-    cv::Mat no_return_mask;
-    cv::inRange(lidar_depth_map_8u, 0, 1, no_return_mask);
-
-    cv::Mat erosion_dst;
-    cv::Mat element = cv::getStructuringElement(
-      cv::MORPH_RECT, cv::Size(2 * erode_kernel_ + 1, 2 * erode_kernel_ + 1),
-      cv::Point(erode_kernel_, erode_kernel_));
-    cv::erode(no_return_mask, erosion_dst, element);
-    cv::dilate(erosion_dst, no_return_mask, element);
-
-    static boost::circular_buffer<cv::Mat> no_return_mask_buffer(buffering_frames_);
-
-    cv::Mat no_return_mask_result(cv::Size(horizontal_bins, vertical_bins), CV_8UC1, cv::Scalar(0));
-    cv::Mat time_series_blockage_mask(
-      cv::Size(horizontal_bins, vertical_bins), CV_8UC1, cv::Scalar(0));
-    cv::Mat no_return_mask_binarized(
-      cv::Size(horizontal_bins, vertical_bins), CV_8UC1, cv::Scalar(0));
+  }
+  full_size_depth_map.convertTo(lidar_depth_map_8u, CV_8UC1, 1.0 / 300);
+  cv::Mat no_return_mask(cv::Size(ideal_horizontal_bins, vertical_bins), CV_8UC1, cv::Scalar(0));
+  cv::inRange(lidar_depth_map_8u, 0, 1, no_return_mask);
+  cv::Mat erosion_dst;
+  cv::Mat blockage_element = cv::getStructuringElement(
+    cv::MORPH_RECT, cv::Size(2 * blockage_kernel_ + 1, 2 * blockage_kernel_ + 1),
+    cv::Point(blockage_kernel_, blockage_kernel_));
+  cv::erode(no_return_mask, erosion_dst, blockage_element);
+  cv::dilate(erosion_dst, no_return_mask, blockage_element);
+  static boost::circular_buffer<cv::Mat> no_return_mask_buffer(blockage_buffering_frames_);
+  cv::Mat time_series_blockage_result(
+    cv::Size(ideal_horizontal_bins, vertical_bins), CV_8UC1, cv::Scalar(0));
+  cv::Mat time_series_blockage_mask(
+    cv::Size(ideal_horizontal_bins, vertical_bins), CV_8UC1, cv::Scalar(0));
+  cv::Mat no_return_mask_binarized(
+    cv::Size(ideal_horizontal_bins, vertical_bins), CV_8UC1, cv::Scalar(0));
 
-    static uint frame_count;
-    frame_count++;
-    if (buffering_interval_ != 0) {
-      no_return_mask_binarized = no_return_mask / 255;
-      if (frame_count == buffering_interval_) {
-        no_return_mask_buffer.push_back(no_return_mask_binarized);
-        frame_count = 0;
-      }
-      for (const auto & binary_mask : no_return_mask_buffer) {
-        time_series_blockage_mask += binary_mask;
-      }
-      cv::inRange(
-        time_series_blockage_mask, no_return_mask_buffer.size() - 1, no_return_mask_buffer.size(),
-        no_return_mask_result);
-    } else {
-      no_return_mask.copyTo(no_return_mask_result);
+  blockage_frame_count_++;
```

Comment:
@badai-nguyen 
There is no upper bound guard on this increment, so it can overflow.

Reply:
@miursh I fixed her https://github.com/autowarefoundation/autoware.universe/pull/3212/commits/0b5b4b7b8b41d7ad5e2be959382763d4656a21ea